src/client/containers/chart-detail-page/chart-detail-page.container.js

```diff
@@ -0,0 +1,215 @@
+import React, { useState, useEffect } from 'react';
+import ChartDetailPage from '../../components/chart-detail-page-layout/chart-detail-page.component';
+import { getTokenWithHeaders } from '../../firebase/getTokenWithHeaders';
+import { ChartDataContext } from './chart-detail-page.context';
+import { useParams } from 'react-router-dom';
+import { dashboardItems } from '../../components/side-navigation/sidebar.component';
+
+function getMaterialFromSlug(slug) {
+  const slugs = dashboardItems.map((item) => item.slug);
+  return dashboardItems[slugs.indexOf(slug)];
+}
+
+const ChartDetailsSmartData = () => {
+  const { materialSlug } = useParams();
+  const material = getMaterialFromSlug(materialSlug);
+  const [logoutModal, setLogoutModal] = useState(false);
+  const [boundaryData, setBoundaryData] = useState({});
+  const [materialName, setMaterialName] = useState(material.value);
+  const [materialId, setMaterialId] = useState(material.id);
+  const [sensorData, setSensorData] = useState([]);
+  const [unit, setUnit] = useState('');
+  const [startDate, setStartDate] = useState('');
+  const [currentDate, setCurrentDate] = useState('');
+  const [stages, setStages] = useState([]);
+  const [selectedChartButtonId, setSelectedChartButtonId] = useState(null);
+  const [activeChartButton, setActiveChartButton] = useState(false);
+  const [startCustom, setStartCustom] = useState('');
+  const [endCustom, setEndCustom] = useState('');
+  const [updateClick, setUpdateClick] = useState(false);
+
+  useEffect(() => {
```

Comment:
you could also write functions for fetching particular data, and then call them in one useEffect